src/backend/utils/mmgr/mcxt.c

```diff
@@ -770,11 +781,49 @@ MemoryContextStats_recur(MemoryContext topContext, MemoryContext rootContext,
 
 	for (child = topContext->firstchild; child != NULL; child = child->nextchild)
 	{
+#ifdef EXTRA_DYNAMIC_MEMORY_DEBUG
+		HTAB *prev_chunk_htable = NULL;
+		HTAB *temp_chunks_htable = NULL;
+		bool is_need_to_print_logs;
+#endif
+
 		/* Get name and ancestry of this MemoryContext */
 		name = MemoryContextName(child, rootContext, nameBuffer, nameBufferSize);
 
+#ifdef EXTRA_DYNAMIC_MEMORY_DEBUG
+		/*
+		   At this case we will get stats of next child,
+		   but after that, we will print stats of previous child.
+		   We must to save chunks_htable to other variable (prev_chunk_htable)
+		   to get correct stats of next child.
+		 */
+		is_need_to_print_logs = (child->firstchild == NULL &&
```

Comment:
Please reread this condition. I think `(child->firstchild == NULL && siblingCount != 0)` already covers `(child->firstchild == NULL && strcmp(name, prevChildName) == 0 && siblingCount != 0)`